tiny_cnn/layers/arithmetic_layer.h

```diff
@@ -51,8 +51,8 @@ class elementwise_add_layer : public layer {
     }
 
     void forward_propagation(cnn_size_t worker_index,
-                             const std::vector<vec_t*>& in_data,
-                             std::vector<vec_t*>& out_data) override {
+                             const vec_t** in_data,
+                             vec_t** out_data) override {
```

Comment:
I am not really in favour of these changes. Do we really have to go into "2 star programming"? One of the reasons I like and use tiny-cnn is because of it's modern, clear code, and if we start introducing these pointers-to-pointers, everything becomes much harder to reason about.

1) Since we're passing around references, I don't see why there would be dynamic allocations in the first place? If there are, can we not solve it in a different way? The memory has to be allocated somewhere, can we not use `::reserve()` there, for example?

2) If this is the only way, can you show a speed difference of these isolated changes, showing that the loss of clarity and ability to reason about the code would be worth it?

3) If we were to adopt these changes, please at least let us add a a clear documentation to each function and the parameters.